src/main/java/sg/ncl/MainController.java

```diff
@@ -2744,6 +2746,101 @@ public String adminTeamUsage(Model model,
+
+        HttpEntity<String> request = createHttpEntityHeaderOnly();
+
+        ResponseEntity responseEntity;
+        try {
+            responseEntity = restTemplate.exchange(properties.getEnergyStatistics("startDate=" + start, "endDate=" + end), HttpMethod.GET, request, String.class);
+        } catch (RestClientException e) {
+            log.warn("Error connecting to sio team service for energy usage: {}", e);
+            redirectAttributes.addFlashAttribute(MESSAGE, ERR_SERVER_OVERLOAD);
+            return "redirect:/energy_usage";
+        }
+
+        String responseBody = responseEntity.getBody().toString();
+        JSONArray jsonArray = new JSONArray(responseBody);
+
+        // handling exceptions from SIO
+        if (RestUtil.isError(responseEntity.getStatusCode())) {
+            MyErrorResource error = objectMapper.readValue(responseBody, MyErrorResource.class);
+            ExceptionState exceptionState = ExceptionState.parseExceptionState(error.getError());
+            switch (exceptionState) {
+                case START_DATE_AFTER_END_DATE_EXCEPTION:
+                    log.warn("Get energy usage : Start date after end date eroor");
+                    redirectAttributes.addFlashAttribute(MESSAGE, ERR_START_DATE_AFTER_END_DATE);
+                    return "redirect:/energy_usage";
+                default:
+                    log.warn("Get energy usage : sio or deterlab adapter connection error");
+                    redirectAttributes.addFlashAttribute(MESSAGE, ERR_SERVER_OVERLOAD);
+                    return "redirect:/energy_usage";
+            }
+        } else {
+            log.info("Get Get energy usage info : {}", responseBody);
+        }
+
+        Map<String, Double> dataToPlotChart = new HashMap<>();
+
+        double sumEnergy = 0.00;
+        List<String> listOfDate = new ArrayList<String>();
+        List<Double> listOfEnergy = new ArrayList<Double>();
+        ZonedDateTime currentZonedDateTime = getZonedDateTime2(start);
+        String currentDate = null;
+        for (int i = 0; i < jsonArray.length(); i++) {
+            sumEnergy  += jsonArray.getDouble(i);
+            currentDate = currentZonedDateTime.format(formatter);
+            listOfDate.add(currentDate);
+            listOfEnergy.add(jsonArray.getDouble(i));
+            currentZonedDateTime = getZonedDateTime2(currentDate).plusDays(1);
+        }
+
+
+        model.addAttribute("listOfDate", listOfDate);
+        model.addAttribute("listOfEnergy", listOfEnergy);
+        //model.addAttribute("dataToPlotChart", dataToPlotChart);
+        model.addAttribute("energy", sumEnergy);
+        return "energy_usage";
+    }
+
+    /**
+     * Get simple ZonedDateTime from date string in the format 'YYYY-MM-DD'.
+     * @param date  date string to convert
+     * @return      ZonedDateTime of
+     */
+    private ZonedDateTime getZonedDateTime2(String date) {
+        if (date != null) {
+            String[] result = date.split("-");
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Replace the type specification in this constructor call with the diamond operator ("<>"). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS2293)